### DISCUSSION TITLE
Url management - regex save to hostsetting

### DISCUSSION THREAD

**Discussion by thienvc** [dnnsoftware/Dnn.Platform#6224] [General]
![image](https://github.com/user-attachments/assets/b7820d60-ac0d-44be-b52a-577e414099b7)

https://github.com/dnnsoftware/Dnn.Platform/blob/7a31fdc9436a5f3b9268cb51b68b73f0c773eb16/Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Services/SeoController.cs#L603-L613

can someone explain to me when (what condition) then PortalID = NULL at line 605
I tried to save to HostSettings but failed, it still saves itself to PortalSettings

**Comment by mitchelsellers** [comment]
All of those settings are global, it will never have a portal id.  (The globe icon is the UI indicator of that).

The processor for storing is shared among other items

**Reply by thienvc** [reply]
You can test change any thing and Save, Nothing changed in `HostSettings`, but effected `PortalSettings`
```
SELECT * FROM [dbo].[HostSettings] where SettingName like '%AUM_%' order by LastModifiedOnDate DESC
SELECT * FROM [dbo].[PortalSettings] where SettingName like '%AUM_%' order by LastModifiedOnDate DESC
```

**Reply by thienvc** [reply]
https://github.com/dnnsoftware/Dnn.Platform/blob/36560afc7fd9f68927b20fbe5b1ba16c5e903d91/Dnn.AdminExperience/Library/Dnn.PersonaBar.Library/PersonaBarApiController.cs#L5-L14

PortalId is not Null

**Reply by valadas** [reply]
In this context Null.NullInteger is -1

**Reply by thienvc** [reply]
Not sure if you guys checked? but I had to go to SQLConsole to update AUM HostSettings, because when I updated from PersonalBar everything was saved to PortalSettings
